File: py/p_2_3.py
import cv2 as cv
import sys

img = cv.imread("./img/face.jpg")
if img is None:
    sys.exit("파일을 찾을수 없습니다")

gray= cv.cvtColor(img,cv.COLOR_BGR2GRAY) # BGB컬러 영상을 명암 영상으로 변환
gray_small= cv.resize(gray,dsize=(0,0),fx=0.5,fy=0.5) # 반으로 축소
cv.imshow("image Display",gray)
cv.imshow("image Display2",gray_small)
cv.waitKey()
cv.destroyAllWindows()

File: src/cv_dl/p2_3.rs
use opencv::{
    core::{self, MatTraitConst, Size_}, highgui, imgcodecs::{self, IMREAD_COLOR}, imgproc::{self, COLOR_BGR2GRAY}, Result
};

pub fn main() -> Result<()> {
    let img = imgcodecs::imread("./img/face.jpg", IMREAD_COLOR)?;
    if img.empty() {
        std::process::exit(0);
    }
    let mut dst= core::Mat::default();
    imgproc::cvt_color(&img, &mut dst, COLOR_BGR2GRAY, 0)?;
    let mut resize_dst= core::Mat::default();

    imgproc::resize(&dst, &mut resize_dst, Size_::from((0,0)), 0.5, 0.5,0 )?;
    highgui::imshow("Image Display", &img)?;
    highgui::imshow("Image Display2", &resize_dst)?;
    highgui::wait_key(0)?;
    highgui::destroy_all_windows()?;
    Ok(())
}
